In [1]:
import sys
sys.path.append(r"C:\Users\hjia9\Documents\GitHub\data-analysis")
sys.path.append(r"C:\Users\hjia9\Documents\GitHub\data-analysis\read")
from read_scope_data import read_trc_data

import numpy as np
import matplotlib.pyplot as plt
from scipy.ndimage import gaussian_filter1d

%matplotlib qt
plt.rcParams.update({'font.size': 14})

In [7]:
probe = 'M3'
data_a = {}
data_b = {}
for ch in [1,2,3,4]:
    channel = 'C' + str(ch)

    ifn = f"C:\data\LAPD\JAN2024_diverging_B\mach-calibration\{channel}-Mach-calibrate-{probe}-a-00000.trc"
    sig, tarr = read_trc_data(ifn)
    data_a[ch] = gaussian_filter1d(sig, 100) / 7.3 # R=7.3 Ohm

    ifn = f"C:\data\LAPD\JAN2024_diverging_B\mach-calibration\{channel}-Mach-calibrate-{probe}-b-00000.trc"
    sig, tarr = read_trc_data(ifn)
    data_b[ch] = gaussian_filter1d(sig, 100) / 7.3 # R=7.3 Ohm

tarr *= 1e3

Reading data...
Done
Reading data...
Done
Reading data...
Done
Reading data...
Done
Reading data...
Done
Reading data...
Done
Reading data...
Done
Reading data...
Done


In [8]:
# Ratio plot with area calibration factors applied
start = 250000
stop = 900000

fig, ax = plt.subplots(2, sharex=True)
ax[1].set_xlabel('Time (ms)')

ax[0].plot(tarr[start:stop], data_a[1][start:stop], label=f'South')
ax[0].plot(tarr[start:stop], data_a[3][start:stop], label=f'North')
ax[0].set_title('Ion saturation current')
ax[0].set_ylabel('Isat (A)')

sig =  (data_a[1] - data_a[3]) / (data_a[1] + data_a[3])
ax[1].plot(tarr[start:stop], sig[start:stop])
ax[1].set_ylabel('ratio')

# print('average ratio:', np.mean(sig[start:stop]))
# print('standard deviation:', np.std(sig[start:stop]))

ax[0].legend(title='facing')
ax[0].grid(True)
ax[1].grid(True)
plt.tight_layout()

In [20]:
# Ratio for area calibration
start = 250000
stop = 900000

fig, ax = plt.subplots(2, sharex=True)
ax[1].set_xlabel('Time (ms)')

ax[0].plot(tarr[start:stop], data_a[1][start:stop], label=f'UL')
ax[0].plot(tarr[start:stop], data_a[2][start:stop], label=f'LL')
ax[0].plot(tarr[start:stop], data_b[4][start:stop], label=f'LR => UL')
ax[0].plot(tarr[start:stop], data_b[3][start:stop], label=f'UR => LL')
ax[0].set_title('South')
ax[0].set_ylabel('Isat (A)')

ax[1].plot(tarr[start:stop], data_a[3][start:stop], label=f'UR')
ax[1].plot(tarr[start:stop], data_a[4][start:stop], label=f'LR')
ax[1].plot(tarr[start:stop], data_b[2][start:stop], label=f'LL => UR')
ax[1].plot(tarr[start:stop], data_b[1][start:stop], label=f'UL => LR')
ax[1].set_title('North')
ax[1].set_ylabel('Isat (A)')
ax[0].legend(loc='lower left')
ax[0].grid(True)
ax[1].grid(True)


sig = data_a[3] / data_a[4]
print('average ratio:', np.mean(sig[start:stop]))
print('standard deviation:', np.std(sig[start:stop]))
sig = data_b[2] / data_b[1]
print('average ratio:', np.mean(sig[start:stop]))
print('standard deviation:', np.std(sig[start:stop]))

average ratio: 0.9987048606821717
standard deviation: 0.13442319160499824
average ratio: 1.1255444302709021
standard deviation: 0.0726259044387001


## M3
Normalized area \
C1 -- UL 1 \
C2 -- LL 1 \
C3 -- UR 1 \
C4 -- LR 1.15

### Side A
Channel C1 to C2 \
average ratio: 1.02 \
standard deviation: 0.016 \
Channel C3 to C4 \
average ratio: 0.87 \
standard deviation: 0.025

### Side B
Channel C1 to C2 \
average ratio: 0.97 \
standard deviation: 0.061 \
Channel C3 to C4 \
average ratio: 0.78 \
standard deviation: 0.015

A1 -- B4 \
average ratio: 0.88 \
standard deviation: 0.031 \
A4 -- B1 \
average ratio: 0.86 \
standard deviation: 0.029

A2 -- B3 \
average ratio: 1.1 \
standard deviation: 0.01 \
B2 -- A3 \
average ratio: 1.01 \
standard deviation: 0.04

## M1
Normalized area \
C1 -- UL 1 \
C2 -- LL 1.14 \
C3 -- UR 1 \
C4 -- LR 1

### South facing
A1--B4 \
average ratio: 1.09 \
standard deviation: 0.339 \
A2--B3 \
average ratio: 1.15 \
standard deviation: 0.345

A1--A2
average ratio: 0.87
standard deviation: 0.032 \
B4--B3 \
average ratio: 0.92 \
standard deviation: 0.068

### North facing
A3--B2 \
average ratio: 0.95 \
standard deviation: 0.071 \
A4--B1 \
average ratio: 1.09 \
standard deviation: 0.162

A3--A4 \
average ratio: 1 \
standard deviation: 0.134 \
B2--B1 \
average ratio: 1.13 \
standard deviation: 0.073